From 6eec2eb7c700947940c02847bdc2a9624897ecf8 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 13 Sep 2016 01:14:25 +0100
Subject: [PATCH 6/9] raven: Use proper timestamp on focus

This is just a minor fix for WM conformancy, relating to issue #583,
making sure we do things by the book, and not sending a 0 timestamp.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 raven/Makefile.am | 1 +
 raven/raven.vala  | 8 +++++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/raven/Makefile.am b/raven/Makefile.am
index e511ad0..8714338 100644
--- a/raven/Makefile.am
+++ b/raven/Makefile.am
@@ -67,6 +67,7 @@ libraven_la_LIBADD = \
 
 libraven_la_VALAFLAGS = \
 	--pkg gtk+-3.0 \
+	--pkg gdk-x11-3.0 \
 	--pkg gio-unix-2.0 \
 	--pkg budgie-1.0 \
 	--pkg toplevel \
diff --git a/raven/raven.vala b/raven/raven.vala
index 1840be0..a9fd1ce 100644
--- a/raven/raven.vala
+++ b/raven/raven.vala
@@ -242,7 +242,13 @@ public class Raven : Gtk.Window
             return;
         }
         if (!has_toplevel_focus) {
-            window.focus(Gdk.CURRENT_TIME);
+            /* X11 specific. */
+            Gdk.Display? display = screen.get_display();
+            if (display is Gdk.X11Display) {
+                window.focus(Gdk.X11Display.get_user_time(display));
+            } else {
+                window.focus(Gtk.get_current_event_time());
+            }
         }
     }
 
-- 
2.10.0

